### ISSUE TITLE
Firstpasswd mails end up in bad/msg spool

### ISSUE THREAD

**Issue by qosobrin** [sympa-community/sympa#1438]
Version
-------
6.2.60

Installation method
-------------------
Debian Bulleye package

Expected behavior
-----------------
A user tries to create a new account using `https://lists.domain.org/wws/renewpasswd/` Then he receives an e-mail with a link to a page to set the password for this new account.

Actual behavior
---------------
A user tries to set his new account's password using `https://lists.domain.org/wws/renewpasswd/` but **never receives the e-mail** message with the link to set the password and generate the account.

Steps to reproduce
------------------
We have a two domain sympa installation, one for `lists.domain.com` and another for `lists.domain.org`. `/etc/sympa/sympa/sympa.conf` has the following information among many other:

```
domain			lists.domain.com
listmaster		listmaster@domain.com
wwsympa_url		https://lists.domain.com/wws
sendmail_aliases	/etc/mail/sympa/aliases
```

`/etc/sympa/lists.domain.com/robot.conf` has the following information among many other:

```
domain			lists.domain.com
listmaster		listmaster@domain.com
wwsympa_url		https://lists.domain.com/wws
sendmail_aliases	/etc/mail/sympa/aliases-lists.domain.com

```
`/etc/sympa/lists.domain.org/robot.conf` has the following information among many other:

```
domain			lists.domain.org
listmaster		istmaster@domain.org
wwsympa_url		https://lists.domain.org/wws
sendmail_aliases	/etc/mail/sympa/aliases-lists.domain.org

```
When a user tries to create an account in lists.domain.com he receives an e-mail with the link to the page to set the password for the new account. But the same does not happen when another user tries to create his account in the other robot, lists.domain.org; he never receives his e-mail message. Further examination shows that the e-mail is sent to the bad/msg spool and never delivered. Setting sympa to `log_level 3` reveals the following entries in `sympa.log` (these logs may be incomplete since they were obtained from a high activity production server and I was not able to identify more lines)

```
Jul 13 10:52:57 hostname wwsympa[50479]: notice Sympa::Spool::Outgoing::store() Message Sympa::Message::Template <sympa@lists.domain.org.1657702377.50479,4971> is stored into bulk spool as <1.5.1657702377.1657702377.201188.sympa@lists.domain.org_s,50479,5250>
Jul 13 10:52:57 hostname bulk[208084]: debug3 Sympa::Spool::unmarshal_metadata(/var/spool/sympa/bulk/pct, 1.5.1657702377.1657702377.201188.sympa@lists.domain.org_s,50479,5250/s, qr<(?^:\A(\w+)\.(\w+)\.(\d+)\.(\d+\.\d+)\.(\@?[^\s\@]*)\@([\w\.\-*]*)_(\w+),(\d+),(\d+)/(\w+)\z)>)
Jul 13 10:52:57 hostname bulk[208084]: debug3 Sympa::Spool::unmarshal_metadata() messagekey=1.5.1657702377.1657702377.201188.sympa@lists.domain.org_s,50479,5250/s, context=*, priority=1
```

Additional information
----------------------
This same or similar behaviour was previously reported in https://listes.renater.fr/sympa/arc/sympa-users/2021-06/msg00011.html

**Comment by ikedas** [comment]
Hi @qosobrin ,

It seems the process `bulk[208084]` that quarantined the message into `bad` directory.  It would log any errors sometime after the time you indicated. Please show the log of that process up to that point: E.g. run "`grep -F 'bulk[208084]' logfile`".

**Comment by qosobrin** [comment]
Hi, @ikedas:

Thank you very much for your prompt answer. Here you have the complete log for that process:

```
Jul 13 10:51:47 hostname bulk[208061]: notice Sympa::Process::daemonize() Starting bulk daemon, PID 208084
Jul 13 10:51:47 hostname bulk[208084]: notice main:: Bulk 6.2.60 Started
Jul 13 10:52:57 hostname bulk[208084]: debug3 Sympa::Spool::unmarshal_metadata(/var/spool/sympa/bulk/pct, 1.5.1657702377.1657702377.201188.sympa@lists.domain.org_s,50479,5250/s, qr<(?^:\A(\w+)\.(\w+)\.(\d+)\.(\d+\.\d+)\.(\@?[^\s\@]*)\@([\w\.\-*]*)_(\w+),(\d+),(\d+)/(\w+)\z)>)
Jul 13 10:52:57 hostname bulk[208084]: debug3 Sympa::Spool::unmarshal_metadata() messagekey=1.5.1657702377.1657702377.201188.sympa@lists.domain.org_s,50479,5250/s, context=*, priority=1
Jul 13 10:53:20 hostname bulk[208084]: notice main::sigterm() Signal TERM received, still processing current task
Jul 13 10:53:20 hostname bulk[208084]: notice main:: Bulk exited normally due to signal
```

Best regards.

**Comment by ikedas** [comment]
Probably it is the same as a report "[\[sympa-users\] incoming message causes SIGTERM which stops Sympa processes](https://listes.renater.fr/sympa/arcsearch_id/sympa-users/2022-07/CAPBxAQ4-sUynUqA5fcJ3kb4xqZZoC5%2BZz7Hr2L0WmaCPn%3Db6Yg%40mail.gmail.com)":  One of sympa services (maybe `sympa_msg.pl`) crashed and Systemd terminated all other services.

Could you please check the contents of `<number>.stderr` files under the `$SPOOLDIR/tmp` directory? As the `<number>` represents the PID, you can check it against the log to see which process may have output the messages in the file.

**Comment by qosobrin** [comment]
Hi, @ikedas:

I could not find a `<number>.stderr` file under the `$SPOOLDIR/tmp` with the number associated to the bulk process. I even checked for files in `$SPOOLDIR/tmp` with creation time around the date and time when the password was requested but with no luck.

Then I requested a new password and waited but no `<number>.stderr` was generated. I repeated again the password request  while grepping `sympa.log` for "Signal TERM received" string also with no luck.

It seems that these firstpasswd mails ended up in bad/msg spool without stopping any sympa process.

Best regards.

**Comment by ikedas** [comment]
Further investigation is impossible due to lack of information.

This issue is closed at this time.  when you will encounter the same event, please submit the issue again.

**Comment by qosobrin** [comment]
Sorry for not being able to provide further information on the issue but none of the files requested were generated when I reproduced the problem again. And I was on vacation when @ikedas closed this issue so I could not carry out more tests on my production server.

Nonetheless, I managed to make my server start sending Firstpasswd mails. I am posting this not for the issue to be investigated but for any other user who can come up with this very same problem.

I will start explaining how I installed my current production server because the problem might be related with it. In my institution we had a Sympa server running the Debian Stretch version of Sympa (6.2.16) with a single mail domain in our lists (#1367). Then we installed a new preproduction server from scratch using Debian Bullseye version of Sympa (6.2.60) and configured it to serve lists in two mail domains: one was the same as the 6.2.16 server and the other one was new. We did our tests in preproduction (even movied production server data into into it) and everything worked fine (oops!, we missed the firstpasswd mail test). So, we cloned this preproduction virtual machine, configured the clone to become the production server, moved our production data into this new machine and did an upgrade of the data:

```
/usr/lib/sympa/bin/sympa.pl --upgrade --from=6.2.16
```

After upgrading the data we started having the problems described in this issue with the new domain (the previous one worked fine in this aspect). We needed to create an internal account for the listmaster of this new domain but we could not receive the Firstpasswd email. So we read the message stored in the `bad` directory of the spool and used the URL provided in that message to create the listmaster account. With this account we were able to create a new list in this domain and this, somehow, solved the problem. Once we created this first list, new generated firstpasswd mails started to be delivered and the problem was solved. I do not know how this happened, but I am sure that it is working now. 

So, if anyone comes into this problem I would recommend to give my _"solution"_ a try.

Thank you very much for the help provided and best regards to all.
